apps/pp/serializers.py

```diff
@@ -23,6 +26,52 @@ class ResourceSerializer(ResourceIdSerializer, ResourceTypeSerializer):
     pass
 
 
+class LinkField(serializers.SerializerMethodField, serializers.URLField):
+    pass
+
+
+# Relation
+
+class ResourceLinksSerializer(serializers.Serializer):
+    self = LinkField()
+    self_link_url_name = None
+
+    def get_self(self, instance):
+        obj_id = getattr(instance, 'id', None) or instance
```

Comment:
Just a thought - we could raise an exception if `self_link_url_name` is not implemented; of course unless there are many more hazardous spots that I haven't seen! ;)